### PR TITLE
metrics setup

### PR DESCRIPTION
**Pull request by shivasishdas**
- Setting up a few basic app metrics
- Setting web metric port to `9000` and worker metric port to `9001`. Refer [PR](https://github.com/razorpay/kube-manifests/pull/1018/files)

### CODE REVIEW

File: internal/subscriber/subscriber.go [hunk 1]
```diff
@@ -157,6 +165,9 @@ func (s *Subscriber) acknowledge(ctx context.Context, req *AckMessage) {
 	}
 
 	removeMessageFromMemory(stats, req.MessageID)
+
+	subscriberMessagesAckd.WithLabelValues(s.topic, s.subscription).Add(1)
+	subscriberTimeTakenToAckMsg.WithLabelValues(s.topic, s.subscription).Observe(time.Now().Sub(msg.PublishTime).Seconds())
```
Comment (Abhishekvrshny) [inline internal/subscriber/subscriber.go:170]: This is diff of the time when the message was published and then acked, right? Should we have a similar metric from publish to consume and from consume to ack/modack?

Comment (Abhishekvrshny) [reply internal/subscriber/subscriber.go:170]: Also, similar metric for modack is missing

File: internal/subscriber/subscriber.go [hunk 2]
```diff
@@ -340,13 +359,15 @@ func (s *Subscriber) Run(ctx context.Context) {
...
 						s.consumedMessageStats[tp].maxCommittedOffset = resp.Offset
+						subscriberMessagesConsumed.WithLabelValues(msg.Topic, s.subscription).Add(1)
 					}
 
 					ackDeadline := time.Now().Add(minAckDeadline).Unix()
 					s.consumedMessageStats[tp].Store(msg, ackDeadline)
 
 					ackID := NewAckMessage(s.subscriberID, msg.Topic, msg.Partition, msg.Offset, int32(ackDeadline), msg.MessageID).BuildAckID()
 					sm = append(sm, &metrov1.ReceivedMessage{AckId: ackID, Message: protoMsg, DeliveryAttempt: 1})
+					subscriberMemoryMessagesCountTotal.WithLabelValues(s.topic, s.subscription).Set(float64(len(s.consumedMessageStats[tp].consumedMessages)))
```
Comment (Abhishekvrshny) [inline internal/subscriber/subscriber.go:370]: this is the current total number of messages in the heap right?

**Comment by Abhishekvrshny** [comment]
should an `env` label be also added? eg `stage`, `prod` etc?

**Comment by Abhishekvrshny** [comment]
Add GRPC request metrics with latency and kafka, consul request metrics with latency

File: internal/node/core.go [hunk 3]
```diff
@@ -30,6 +31,11 @@ func NewCore(repo IRepo) *Core {
 
 // CreateNode creates a new node
 func (c *Core) CreateNode(ctx context.Context, m *Model) error {
+	nodeOperationCount.WithLabelValues(env, "CreateNode").Inc()
+
+	startTime := time.Now()
+	defer nodeOperationTimeTaken.WithLabelValues(env, "CreateNode").Observe(time.Now().Sub(startTime).Seconds())
```
Comment (sunnyagg) [inline internal/node/core.go:37]: time in ms than secs

Comment (shivasishdas) [reply internal/node/core.go:37]: Going by this : https://prometheus.io/docs/practices/naming/
> ...should use base units (e.g. seconds, bytes, meters - not milliseconds, megabytes, kilometers). See below for a list of base units.

File: internal/publisher/core.go [hunk 4]
```diff
@@ -55,6 +55,7 @@ func (p *Core) Publish(ctx context.Context, req *metrov1.PublishRequest) ([]stri
 		}
 		msgIDs = append(msgIDs, msgResp.MessageID)
 	}
+	publisherMessagesPublished.WithLabelValues(env, req.Topic).Add(float64(len(msgIDs)))
```
Comment (sunnyagg) [inline internal/publisher/core.go:58]: why is this float ?

**Review by sunnyagg — APPROVED** [review]

File: internal/publisher/core.go (hunk 4 above)
Comment (shivasishdas) [reply internal/publisher/core.go:58]: The function expects `float64` 

```
// To create Counter instances, use NewCounter.
type Counter interface {
	Metric
	Collector

	// Inc increments the counter by 1. Use Add to increment it by arbitrary
	// non-negative values.
	Inc()
	// Add adds the given value to the counter. It panics if the value is <
	// 0.
	Add(float64)
}
```

File: internal/node/core.go (hunk 3 above)
Comment (shivasishdas) [reply internal/node/core.go]: changed

**Review by sunnyagg — APPROVED** [review]

**Pull request merged**